src/scales/quantitative.js

```diff
@@ -151,8 +151,10 @@ export function ScaleQuantize(key, channels, {
   range,
   reverse
 }) {
-  domain = extent(domain); // TODO preserve descending domains?
+  const descending = order(domain) < 0; // preserve descending domain
```

Comment:
I’m not sure we can assume that domain is an array here. I believe it’s allowed to be an iterable, so we’d need to arrayify it first. Also, would it be simpler to flip `reverse` here if the `domain` is decending, rather than reversing the thresholds later? Then we won’t need to support descending thresholds for reversed scales generally.

```js
  if (order(domain) < 0) reverse = !reverse; // preserve descending domain
```

Reply:
To answer my own question here, flipping `reverse` would produce a different result: it flips the range of the scale rather than the domain. So, you’d get this:

<img width="263" alt="Screen Shot 2022-03-25 at 12 50 54 PM" src="https://user-images.githubusercontent.com/230541/160191569-c1c7f6bf-f2ab-408e-8b95-ed542dd52ad7.png">

Instead of this:

<img width="263" alt="Screen Shot 2022-03-25 at 12 50 35 PM" src="https://user-images.githubusercontent.com/230541/160191528-6a13e288-eb10-4424-8a88-81a5d60f9d5d.png">

The behavior of the scale is the same, but the appearance of the legend is different. If we want to be consistent with descending domains on continuous scales, then we want to flip the domain rather than the range.